_includes/pbc/all/install-features/202307.0/install-the-product-rating-and-reviews-feature.md

```diff
@@ -1,201 +1,1212 @@
+{
+    /**
+     * @return array<\Spryker\Zed\SynchronizationExtension\Dependency\Plugin\SynchronizationDataPluginInterface>
+     */
+    protected function getSynchronizationDataPlugins(): array
+    {
+        return [
+            new ProductReviewSynchronizationDataPlugin(),
+        ];
+    }
+}
+```
+
+6. Set up event trigger feature:
+
+| PLUGIN                                            | SPECIFICATION                                                                    | PREREQUISITES | NAMESPACE                                                  |
+|---------------------------------------------------|----------------------------------------------------------------------------------|---------------|------------------------------------------------------------|
+| ProductReviewEventResourceQueryContainerPlugin    | Allows synchronizing the product review search table content with Elasticsearch. | None          | Spryker\Zed\ProductReviewSearch\Communication\Plugin\Event |
+
+**src/Pyz/Zed/EventBehavior/EventBehaviorDependencyProvider.php**
 
 ```php
 <?php
 
+namespace Pyz\Zed\EventBehavior;
+
+use Spryker\Zed\EventBehavior\EventBehaviorDependencyProvider as SprykerEventBehaviorDependencyProvider;
+use Spryker\Zed\ProductReviewSearch\Communication\Plugin\Event\ProductReviewEventResourceQueryContainerPlugin;
+
+class EventBehaviorDependencyProvider extends SprykerEventBehaviorDependencyProvider
+{
     /**
-     * @method \Pyz\Yves\ProductReview\ProductReviewFactory getFactory()
+     * @return array<\Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface>
      */
-    class SubmitController extends Pyz\Yves\Application\Controller\AbstractController
+    protected function getEventTriggerResourcePlugins(): array
     {
-        /**
-         * @param \Symfony\Component\HttpFoundation\Request $request
-         */
-        public function submitAction(Request $request)
-        {
-            $customerReference = $this->getFactory()->getCustomerClient()->getCustomer()->getCustomerReference();
-
-            $this->getFactory()->getProductReviewClient()->submitCustomerReview(
-                (new ProductReviewRequestTransfer())
-                        ->setCustomerReference($customerReference)
-                        ->setLocaleName($this->getLocale())
-                        ->setIdProductAbstract($request->attributes->get('idProductAbstract'))
-                        ->setSummary($request->attributes->get('summary'))
-                        ->setDescription($request->attributes->get('description'))
-                        ->setRating($request->attributes->get('rating'))
-                        ->setNickname($request->attributes->get('nickname'))
-            );
-        }
+        return [
+            new ProductReviewEventResourceQueryContainerPlugin(),
+        ];
     }
+}
 ```
 
-### Displaying an Average Rating
+7. Register the event subscriber:
```

Comment:
Same - is there an official confirmation that his name is spelled correctly?

Reply:
This is an old way to listen to the events. But we don't have a new way implemented. Reference to the old way:
https://docs.spryker.com/docs/scos/dev/back-end-development/data-manipulation/event/listen-to-events.html#listen-to-events-with-direct-listeners